lib/beaker/options/parser.rb

```diff
@@ -67,29 +60,39 @@ def split_arg arg
       #                         no .rb files are found overall
       def file_list(paths)
         files = []
-        if not paths.empty?
+        unless paths.empty?
           paths.each do |root|
+            @validator.validate_path(root)
+
+            path_files = []
             if File.file?(root)
-              files << root
+              path_files << root
             elsif File.directory?(root) #expand and explore
-              discover_files = Dir.glob(
-                File.join(root, "**/*.rb")
-              ).select { |f| File.file?(f) }
-              if discover_files.empty?
-                parser_error "empty directory used as an option (#{root})!"
-              end
-              files += discover_files.sort_by {|file| [file.count("/"), file]}
-            else #not a file, not a directory, not nothin'
-              parser_error "#{root} used as a file option but is not a file or directory!"
+              path_files = Dir.glob(File.join(root, '**/*.rb'))
+                               .select { |f| File.file?(f) }
+                               .sort_by { |file| [file.count('/'), file] }
             end
+
+            @validator.validate_files(path_files, root)
+            files += path_files
           end
         end
-        if files.empty?
-          parser_error "no .rb files found in #{paths.to_s}"
-        end
+
+        @validator.validate_files(files, paths.to_s)
```

Comment:
Would it be sufficient for this to be the only call to `validate_files`? It seems like it throws an exception, and all we've done is add files here.

Another approach (I don't know without deeper spelunking if that method is used outside the files in this PR) would be to simply do the empty check that the method does, here inline. Is an exception needed, or could we pass back an empty result and expect callers to do the right thing?